### PR TITLE
Geometry altitude documentation

### PR DESCRIPTION
**Pull request by signedav**
Documentation for:
https://github.com/QGEP/datamodel/pull/90
and
https://github.com/QGEP/QGEP/pull/468

### CODE REVIEW

**Comment by urskaufmann** [comment]
The reach point levels have  nothing to do with the cover level.  If a reach point snaps to a wastewater structure, the altitude can be taken from the wastewater node bottom level (never from the cover). So the example with the two images is wrong

**Comment by signedav** [comment]
Hmm... thanks for the feedback @urskaufmann

So the reach point level (and with it the z value of the reach end/start point of the reach) should not be set by the z value of the snapped feature like we defined, because this would be the cover level since the geometry of the wastewaterstructure "is" the geometry of the cover?

So it should be the `bottom_level` on snapping to a wastewaterstructure, right?

But don't we snap to other objects as well? What value should be taken then? Still Z if no `bottom_level` exists?

**Comment by urskaufmann** [comment]
Yes. This snapping is not just a geometric snapping. Its mainly a (hydraulic) connection from reach point to a wastewater network element (the value that is in fk_wastewater_networkelement of the reach point). If the fk_wastewater_networkelement is a reach, its not obvious with the reach point level (-> Null). If the fk_... is a node, then it is often a good value to take the bottom level of the connected node. If there is no bottom level -> Null.

**Comment by lbartoletti** [comment]
Thank you @signedav for writing this documentation.
Given the complexity of altitude management, I think it would be necessary to write for the user all the cases you have listed in your PR https://github.com/QGEP/datamodel/pull/90

cc @kandre @ponceta

**Comment by signedav** [comment]
@lbartoletti I tried that summarized with only the info important for an end user in the `note` parts of my changes. Do you mean something else? Or is it just not detailed enough?

Btw. what do you think about the inputs of @urskaufmann . Should we change this solution https://github.com/QGEP/QGEP/pull/468 ?

**Comment by lbartoletti** [comment]
@signedav For me, it's a good documentation. But, on complicated cases like this, I'm still wondering if we shouldn't add a note describing the process as you explained it so well in the other PR.

And yes, I agree with @urskaufmann. @kandre @ponceta What is your opinion?

**Comment by lbartoletti** [comment]
The best thing would still be to be able to visualize the networkelement and determine the different reach points, but this is another development :)

**Comment by signedav** [comment]
@lbartoletti I added some more information to the `note` regarding the input situation, what should illuminate the user better to the behavior. What do you think? It's actually containing the use cases like in the other PR. I wouldn't adapt it "exactly" from the other PR because it's getting more technical then in my opinion.

File: en/user-guide/digitizing/digitizingwastewaterstructures.rst [hunk 1]
```diff
@@ -81,4 +81,4 @@ Geometry synchronization
...
 The geometry of the wastewater structure feature corresponds to the one of the cover. This means, the Z geometry of the digitized feature is actually the entered value on `co_level`.
 
-.. note:: If a cover level changes, the Z value of the cover's geometry will be ajusted. When the geometry changes, the level is ajusted as well. If both value change, the level dominates. The same situation is on editing the wastewater node directly. On this main layer wastewater structure, the only way to set the Z value of the geometry is to set the levels of cover and wastewater node.
\ No newline at end of file
+.. note:: If a cover level changes, the Z value of the cover's geometry will be ajusted. When the geometry changes, the level is ajusted as well.  If both values change, the level takes precendence. The same situation is on editing the wastewater node directly. On this main layer wastewater structure, the only way to set the Z value of the geometry is to set the levels of cover and wastewater node.
```
Comment (ponceta) [inline en/user-guide/digitizing/digitizingwastewaterstructures.rst]: set `co_level` for cover and `wn_bottom_level` for wastewater node.

File: en/user-guide/digitizing/digitizingchannels.rst [hunk 2]
```diff
@@ -77,8 +77,20 @@ When a line object is digitized a series of steps take place in the background i
...
 
 
+Geometry synchronization
+------------------------
+
+The start- and the end-point of the added feature's geometry defines the reach point's geometries. The altitude (Z value of geometry) of the added feature's start- and the end-point is defined by the level values `rp_from_level` and `rp_to_level`. This means, the Z values of the reach point's geometry is set by the level as well. These values could be filled up by the snapped features:
+
+.. figure::images/reach_geometry_snapping_01.jpg
+
+The snapped features 9405 and 9390 have the altitude from the cover levels: 424.090 and 422.900. These values are entered into the attributes `rp_from_level` and `rp_to_level` and can be eddited there by the user. When the user changes these values, the Z value of the geometry of the digitized feature and its reach points will be changed as well.
```
Comment (ponceta) [inline en/user-guide/digitizing/digitizingchannels.rst]: Typo : edited

**Comment by signedav** [comment]
Thanks @ponceta @lbartoletti

**Comment by ponceta** [comment]
> The reach point levels have nothing to do with the cover level. If a reach point snaps to a wastewater structure, the altitude can be taken from the wastewater node bottom level (never from the cover). So the example with the two images is wrong

A user should not snap on covers, in case of overlapping features, **this should be adressed through snapping priorizations.**

A user should be able to snap on **any 3D object and get the Z Value** at INSERTION only. (Construction point, existing wastewaternode, existing reach point) like standard QGIS mechanics.

_In the end, it should be possible to snap to the middle of a 3d Vertex and extract at insertion the estimated altimetry, but of course that's out of scope yet._

**Comment by ponceta** [comment]
@signedav @lbartoletti Looks good to me

**Pull request merged**